src/Compilers/CSharp/Portable/Binder/ContextualAttributeBinder.cs

```diff
@@ -64,5 +66,14 @@ internal Symbol AttributeTarget
                 return _attributeTarget;
             }
         }
+
+        protected override CSDiagnosticInfo SingleTypeNameNotFound(string whereText, DiagnosticBag diagnostics, Location location, AssemblySymbol forwardedToAssembly)
+        {
+            if (!whereText.EndsWith("Attribute"))
```

Comment:
Is this check actually correct?

Consider `[XmlAttribute]` - that looks for `XmlAttribute` and `XmlAttributeAttribute` - the latter of which exists. Given that the compiler looks for both, I'd expect both to be in the failure text.